analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/LocksReleasedAllPathsBase.cs

```diff
@@ -156,5 +167,43 @@ private ProgramState RemoveLock(SymbolicContext context, ISymbol symbol)
 
         private static ISymbol FirstArgumentSymbol(IInvocationOperationWrapper invocation) =>
             IArgumentOperationWrapper.FromOperation(invocation.Arguments.First()).Value.TrackedSymbol();
+
+        protected sealed class LockAcquireReleaseCollector
+        {
+            private static readonly HashSet<string> LockTypes = new() { "Mutex" }; // For some APIs ctor can directly acquire the lock (e.g. Mutex).
```

Comment:
Does this already need a hashset already?  A single string should perform way better. Even an array could be a bit faster by not computing the hash code.